internal/api/handlers.go

```diff
@@ -130,6 +141,17 @@ func GetRecommendationSet(c echo.Context) error {
 
 	var unitChoices = make(map[string]string)
 
+	k8sUnits := c.QueryParam("k8s-units")
+	var setk8sUnits bool
+
+	if k8sUnits == "" || k8sUnits == "true" {
+		setk8sUnits = true
+	} else if k8sUnits == "false" {
+		setk8sUnits = false
+	} else {
+		return c.JSON(http.StatusBadRequest, echo.Map{"status": "error", "message": "invalid option for k8s-units"})
+	}
```

Comment:
This can be moved to function as its getting repeated twice?

Reply:
I don't quite see the advantage, since we don't have any other endpoints

Reply:
okay, maybe I would let @patilsuraj767 and @kgaikwad add their thoughts as well?
more about DRY: https://www.geeksforgeeks.org/dont-repeat-yourselfdry-in-software-development/ there are multiple refs about it, however I felt this post is bit clear

Reply:
We'll also need to modularize [this](https://github.com/RedHatInsights/ros-ocp-backend/blob/main/internal/api/handlers.go#L161-L193) in that case, let's plan it for later when we start work on the plots endpoint.

Reply:
> maybe I would let @patilsuraj767 and @kgaikwad add their thoughts as well?

+1 from my side for extracting duplicate code blocks to function.

@saltgen - Yes, there are definitely changes required to support individual usage(CPU/Mem) units based on `k8sUnits` param selection which are currently missing here. But at least I don't understand why do you want to wait for ^^ to extract out this code block.

Reply:
@kgaikwad I'm not in favor of wrapping almost all logic in functions because function calls have a performance overhead. I believe we should try to achieve a balance.

1. I think we are taking DRY too literally here. We have exactly two usages of the this block of code, because there are two endpoints.

2. We might have the box plots data endpoint soon, then we could plan this block and individual usage(CPU/Mem) units in one go and make something like [this](https://github.com/RedHatInsights/ros-ocp-backend/blob/main/internal/api/utils.go#L67).

Let me know what I can do to unblock this PR.

Reply:
> I'm not in favor of wrapping almost all logic in functions because function calls have a performance overhead.

@saltgen,  When I see this function `GetRecommendationSetList` - https://github.com/RedHatInsights/ros-ocp-backend/blob/1920484c0f5de14bf1aaf8dd094ed679b82035ac/internal/api/handlers.go#L15-L158
Not each logic wrapped into multiple functions and suggestion provided by @upadhyeammit above didn't talk about wrapping each & every logic inside functions.
My vote to the suggestion is purely for code readability and maintenance perspective. In fact, I see repetitive logic at multiple places in this same file which can be extracted into a single function. I want to understand your concern here - Could you please elaborate what kind of performance overhead you are anticipating here with this suggestion? Will it be possible to pass on few references for my understanding?

> We might have the box plots data endpoint soon

Could you provide reference ticket to get when it is planned for and how these both things are correlated?

Reply:
Moving conversation to internal DM